### ISSUE TITLE
fine tuning questions using tiny model 

### ISSUE THREAD

**Issue by cchen028** [jumon/whisper-finetuning#12]
First of all, really appreciate you sharing this. This is my first python project ever but was able to get a model out thanks to your clear instructions and commenting. 

Questions:
1. Is there a reason why the prepared data is prepared in 30 seconds chunk rather than splitting by each line of the provided SRT file? Reason asking is splitting at 30 seconds often cuts off in the middle of a sentence and could cause a bit of in accuracy in the beginning and end whereas each line of SRT usually ends with a short pause.

2. The fine-tuned model transcribes significantly worse then the original model (my base was tiny model for Mandarin using ~10k files to fine tune). I see in your other comments that adjusting the hyper parameters would be the way to go. I did some research but wasn't able to grasp the exact meaning of this, could you give a quick example of "adjusting the hyper parameters" using this project?

**Comment by jumon** [comment]
Thank you for giving this project a try, and I apologize that it didn't work well for you. Here are my responses to your questions:

1. The reason for this is that Whisper is trained in this manner, as far as I understand. When audio is divided in the middle of a sentence, Whisper only generates output before the sentence, shifts to the beginning of the sentence, and starts predicting from there. For more details, please refer to the original Whisper paper. Here's a quote from the paper:
```
When a final transcript segment is only partially included in the current 30 second audio chunk, we predict only its start time token for the segment when in timestamp mode, to indicate that he subsequent decoding should be performed on an audio window aligned with that time, otherwise we truncate the audio to not include the segment.
```

2. In such cases, what I usually do is adjust the learning rate and batch size, along with the gradient accumulation step. Additionally, if you haven't already, I recommend trying the `--train-only-decoder` parameter, as it often stabilizes the training. However, it is also highly possible that there are issues in my code. This project is still a work in progress, and I currently lack the time to further develop it.
